File: 0441-arranging-coins/0441-arranging-coins.py
class Solution:

    # given the number that represents a row, return overal how many coins will be there
    def getCoins(self, num) -> int:
        if num%2 == 0:
            return (num//2) * (num+1)
        return num*((num+1)//2)

    def arrangeCoins(self, n: int) -> int:

        left = 0
        right = n

        while left<=right:
            mid = left + (right-left)//2
            coins = self.getCoins(mid)
            if coins == n:
                return mid
            
            if coins > n:
                right = mid - 1
            else:
                left = mid + 1

        return right
        

File: 0441-arranging-coins/0441-arranging-coins.cpp
class Solution {
public:
    
    long getCoins(int num) {
        return (long)(num)*(num+1)/2;
    }

    int arrangeCoins(int n) {
        
        int left = 0;
        int right = n;

        while(left<=right){

            int mid = left + (right-left)/2;
            long coins = getCoins(mid);

            if(coins==n){
                return mid;
            }

            if(coins>n){
                right = mid - 1;
            }else{
                left = mid + 1;
            }


        }

        return right;

    }
};